source/extensions/filters/http/ext_proc/processor_state.h

```diff
@@ -29,7 +29,7 @@ class QueuedChunk {
   bool end_stream = false;
   // True if the chunk was actually sent to the gRPC stream
   bool delivered = false;
-  Buffer::OwnedImpl data;
```

Comment:
Storing a copy of the data per queue-element is the source of the memory wasting problem.  If ext_proc gRPC is slow, and this queue grow large, it may cause significant memory consumption.  Keeping an aggregated receiving buffer, and just maintaining a logic view of the received data in the queue is the proposed solution.